src/wazuh_db/wdb_global.c

```diff
@@ -599,6 +599,113 @@ int wdb_global_update_agent_group(wdb_t *wdb, int id, char *group) {
     }
 }
 
+int wdb_global_update_agent_groups_hash(wdb_t* wdb, int agent_id, char* groups_string) {
+    cJSON* j_agent_info = NULL;
+    char* agent_groups = NULL;
+    char truncated_groups_hash[9] = {0};
+    os_sha256 groups_hash;
+    int result = OS_INVALID;
+
+    // If the comma separated groups string is not send, read it from 'group' column
+    if(!groups_string) {
+        j_agent_info = wdb_global_get_agent_info(wdb, agent_id);
+        if(!j_agent_info) {
+            mdebug1("Unable to get the agent's '%d' info to update the groups_hash column", agent_id);
+            return OS_INVALID;
+        }
+
+        agent_groups = cJSON_GetStringValue(cJSON_GetObjectItem(j_agent_info->child, "group"));
+        if(!agent_groups) {
+            mdebug2("Empty group column for agent '%d', groups_hash column won't be updated", agent_id);
```

Comment:
I would write it in this way... `...for agent '%d'. The groups_hash column...`

Reply:
Done!